account_payment_term_extension/models/account_payment_term.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2013-2016 Camptocamp SA (Yannick Vaucher)
+# Copyright 2004-2016 Odoo S.A. (www.odoo.com)
+# Copyright 2015-2016 Akretion
+# (Alexis de Lattre <alexis.delattre@akretion.com>)
+# Copyright 2018 Simone Rubino - Agile Business Group
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import calendar
+from functools import reduce
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, exceptions, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_is_zero, float_round
+
+
+class AccountPaymentTermHoliday(models.Model):
+    _name = "account.payment.term.holiday"
+    _description = "Payment Term Holidays"
+
+    payment_id = fields.Many2one(comodel_name="account.payment.term")
+    holiday = fields.Date(required=True)
+    date_postponed = fields.Date(string="Postponed date", required=True)
+
+    @api.constrains("holiday", "date_postponed")
+    def check_holiday(self):
+        for record in self:
+            if fields.Date.from_string(
+                record.date_postponed
+            ) <= fields.Date.from_string(record.holiday):
+                raise ValidationError(
+                    _("Holiday %s can only be postponed into the future")
+                    % record.holiday
+                )
+            if (
+                record.search_count(
+                    [
+                        ("payment_id", "=", record.payment_id.id),
+                        ("holiday", "=", record.holiday),
+                    ]
+                )
+                > 1
+            ):
+                raise ValidationError(
+                    _("Holiday %s is duplicated in current payment term")
+                    % record.holiday
+                )
+            if (
+                record.search_count(
+                    [
+                        ("payment_id", "=", record.payment_id.id),
+                        "|",
+                        ("date_postponed", "=", record.holiday),
+                        ("holiday", "=", record.date_postponed),
+                    ]
+                )
+                >= 1
+            ):
+                raise ValidationError(
+                    _("Date %s cannot is both a holiday and a Postponed date")
+                    % record.holiday
+                )
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    amount_round = fields.Float(
+        string="Amount Rounding",
+        digits="Account",
+        # TODO : I don't understand this help msg ; what is surcharge ?
```

Comment:
Any chance this `TODO` comment cant be resolved?

Reply:
Remove that part of the help message that doesn't apply here. I think it's copied from a field in Odoo (from pricelists).